### ISSUE TITLE
"My Recent Uploads" empty, spinning forever

### ISSUE THREAD

**Issue by nicolas-raoul** [commons-app/apps-android-commons#1393]
Since updating the app.
![screenshot_20180331-192104](https://user-images.githubusercontent.com/99590/38162495-11c90642-351d-11e8-94d4-31fc43a095d4.png)

**Comment by 97balakrishnan** [comment]
I am not having this issue on the master branch's latest version, Can you please specify in which specific case this happens? My recent uploads are working fine on my phone Android 6.0.1

**Comment by misaochan** [comment]
Hmm. Does this issue persist for you, @nicolas-raoul , or is it only the first time?

Thanks for the crash report @97balakrishnan !

**Comment by VojtechDostal** [comment]
This happened to me too after I updated to 2.7. Recent uploada fail to load for me

**Comment by VojtechDostal** [comment]
Interesting find: it does not load no matter how I click between items in the sidebar. But when I change the number of "recent uploads shown" in settings and then go to recent uploads, the problem disappears. However it appears again when I close and reopen the app

**Comment by misaochan** [comment]
@VojtechDostal this is your 2fa account, right? Could you please try logging out and in again and see if that fixes the problem?

Also, does anyone face this problem with a non-2fa account?

**Comment by VojtechDostal** [comment]
Nope I disabled 2FA a few months ago.

I tried signing off and on again and it didnt help. Actually the spinning loading wheel disappears altogether, just blank window now.

All the other functions seem to work.

**Comment by VojtechDostal** [comment]
Sorry the blank window was just before the pictures loaded. So it took a while but now the recent uploads show normally. So it could have been the signing off/on that helped

**Comment by misaochan** [comment]
Thanks for the details @VojtechDostal ! If it happens again, it would be great if logs could be provided (immediately after you see that happening, go to Settings > Send logs to developers).

@maskaravivek Do you think you could take a look at this? I think the authentication might be buggy in this aspect. :)

**Comment by nicolas-raoul** [comment]
I don't use 2FA either.
Yesterday I reduced the number of "recent uploads shown" to 3, but that did not help.
I got the images to show once today, by just opening the app, but now they do not show anymore.

**Comment by nicolas-raoul** [comment]
I have never used 2FA for Wikimedia.
Here is my log after rebooting my phone, opening the Commons app, and letting it spin for a few minutes:
[logs.txt](https://github.com/commons-app/apps-android-commons/files/1867221/logs.txt)

**Comment by misaochan** [comment]
Hm, strange, the logs.txt is only a few lines and none related to the app:

`04-02 14:22:58.643  3434  3445 W art     : Suspending all threads took: 8.916ms04-02 14:22:58.700  3434  3445 I art     : Background partial concurrent mark sweep GC freed 28740(1432KB) AllocSpace objects, 8(160KB) LOS objects, 39% free, 7MB/12MB, paused 17.749ms total 214.824ms04-02 14:22:58.916  3434  3434 I AppCompatViewInflater: app:theme is now deprecated. Please move to using android:theme instead.04-02 14:22:59.478  3434  3434 I Choreographer: Skipped 52 frames!  The application may be doing too much work on its main thread.04-02 14:23:00.434  3434  3434 I Choreographer: Skipped 57 frames!  The application may be doing too much work on its main thread.`

I wonder if there is an issue with how we are collecting the logs... :/

@nicolas-raoul , thanks for the details. Does logging out and back in again fix the issue for you?

**Comment by nicolas-raoul** [comment]
@misaochan Apparently that fixed the problem for @VojtechDostal but I am worried that many other users are affected, so how about using me to test solutions?

I almost never look at my own uploads, so I don't mind. I can also test any APK you might want to try.

**Comment by misaochan** [comment]
@nicolas-raoul The problem is that we're not sure if it actually did work for Vojtech or if it was just coincidence. :) If you could confirm that it solves the problem for you, then at least we can be quite sure that the issue is related to changes that we made to authentication.

Thanks, we will ping you when we have a solution ready to try. :)

**Comment by misaochan** [comment]
Edit: Oh, I see, you mean that you are afraid that you can't reproduce the bug any more once you log out? Good point... :/

**Comment by misaochan** [comment]
In that case, could we please get a logcat for the time when you open the app? Unfortunately the logs option in Settings seems to be bugged, so getting it from Android Studio would be ideal.

**Comment by nicolas-raoul** [comment]
Logging out and in solved the problem.
The problem does not happen anymore even if I restart the app.

**Comment by Jatin0312** [comment]
The problem of continuous spinning and come when I clear data for commons app. It is resolved as mentioned here by logging out and then login.

**Comment by Jatin0312** [comment]
Shouldn't it be log out while clearing app data.

**Comment by misaochan** [comment]
Interesting. Did you guys not have to log in after updating the app? Did it actually save your previous login session despite the update or clearing data?

**Comment by VojtechDostal** [comment]
I dont remember ever logging in after updates. I am always logged in.

**Comment by nicolas-raoul** [comment]
Updating from Play Store (or from an APK with the same signature I guess) does not require re-entering my password.

**Comment by neslihanturan** [comment]
I become able to reproduce this bug on a emulator API level 23, on LogCat debug I saw 
`04-05 16:03:09.986 3372-3372/fr.free.nrw.commons.debug E/SessionManager: User is not logged in`
Maybe works for your finding solution @maskaravivek

**Comment by neslihanturan** [comment]
What about logging out user programatically if we can't find any solution by this weekend? Would it be annoying for users?

**Comment by misaochan** [comment]
Yeah, that's an option that I'm considering, too. However, since this bug is only in beta (not production), I think it is OK to wait a couple more days for Vivek to get better, if needed. No need to rush for a solution.

**Comment by dinaraassan** [comment]
I faced the same problem. 
Interesting fact: "My Recent Uploads" will spinning forever, until I change the screen orientation from vertical to horizontal. In the landscape orientation I can view my recent uploads and once I will change orientation again to the vertical I will still be able to view them until I will close the app. Next time I will open the app, "My Recent Uploads" will spinning forever until I will repeat my manipulations with the orientations. 

I stayed logged in and didn't logged out, my application version upgrade automatically from 2.6.7 to 2.7.0

**Comment by nicolas-raoul** [comment]
I guess changing orientation sort of reloads the view or something?

**Comment by neslihanturan** [comment]
But then our assumption that is related with being logged out or seeming like logged out should be wrong or incomplete. Weirdly @maskaravivek has already solved this issue with that assumption

**Comment by maskaravivek** [comment]
@neslihanturan I have added checks based on the logs that you posted above. 

```
04-05 16:03:09.986 3372-3372/fr.free.nrw.commons.debug E/SessionManager: User is not logged in
```

As i have never been able to reproduce the issue on my end(i haven't tried the latest set of steps shared by you), I can't confidently say if the PR solves the issue.

**Comment by misaochan** [comment]
Hi @nicolas-raoul , this should have been fixed in 2.7.1, can you please help us test if it works for you?

Steps:
1. Unregister from beta and download the production version (2.6.7)
2. Log in to the app
3. Register to beta and update to 2.7.1
4. See if My Recent Uploads works for you?

**Comment by domdomegg** [comment]
@nicolas-raoul is this fixed?

**Comment by nicolas-raoul** [comment]
@domdomegg Sorry I have not tried recently.

**Comment by domdomegg** [comment]
Think this was fixed in #1454, reopen if issue persists with same logs.
